### PR TITLE
WIP - Creating interfaces and services for writes

### PR DESCRIPTION
**Pull request by corylanou**
- [ ] Name writer interfaces
- [ ] Refactor write endpoint
- [ ] Create binary write endpoint

### CODE REVIEW

File: data/writers.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	ReplicaAckQuorum
+	ReplicaAckOwner
+	ReplicaAckAll
+)
+
+const defaultReadTimeout = 5 * time.Second
+
+var ErrTimeout = errors.New("timeout")
+
+type Payload struct {
```
Comment (pauldix) [inline data/writers.go]: Wouldn't it be better to just call this a WriteRequest?

**Comment by pauldix** [comment]
Not sure I understand what an `ackPolicy` is. Is that supposed to be the `ConsistencyLevel` requested?

**Comment by corylanou** [comment]
`ackPolicy` is supposed to mirror `replica acknowledged`.  A terminology that MongoDB already uses.  And yes, it has to do with the consistency write level being asked for.  We probably need to closer align the naming now that I see it.

**Comment by corylanou** [comment]
PR based off of master, needs to be based off of `alpha1`

**Comment by pauldix** [comment]
I'm not familiar with Mongo's terminology, but our design is closer to Cassandra so it makes sense to use their terms. Although I haven't fully digested this so I don't yet have strong opinions, just asking questions to help myself understand :)

**Pull request closed without merging**